### PR TITLE
Make contact required [PD-2568]

### PR DESCRIPTION
**Pull request by tdphillips**
This fixes the immediate problem. I just want to get some feedback on if this is how we want to solve it.

### CODE REVIEW

File: mypartners/forms.py [hunk 1]
```diff
@@ -358,6 +358,9 @@ def __init__(self, *args, **kwargs):
         if 'partner' in kwargs:
             partner = kwargs.pop('partner')
         super(ContactRecordForm, self).__init__(*args, **kwargs)
+        self.fields['contact'].required = True
```
Comment (tdphillips) [inline mypartners/forms.py]: Contact should generally be required.

File: mypartners/forms.py [hunk 2]
```diff
@@ -358,6 +358,9 @@ def __init__(self, *args, **kwargs):
...
+        if not hasattr(self.instance, 'contact'):
```
Comment (tdphillips) [inline mypartners/forms.py]: Unless self.instance.contact doesn't exist (the contact was deleted).

File: mypartners/forms.py [hunk 3]
```diff
@@ -444,7 +447,12 @@ def save(self, request, partner, commit=True):
         PRMAttachment.objects.filter(
             pk__in=self.cleaned_data.get('attach_delete', [])).delete()
 
-        identifier = instance.contact.name
+        if instance.contact:
```
Comment (tdphillips) [inline mypartners/forms.py]: Do some checking to determine what the contact identifier should be.

I'm open to suggestions on the default.

Comment (JLMcLaugh) [inline mypartners/forms.py]: This looks more complete than what we're logging elsewhere. We record the contact id and the __unicode__ value also, so it should be fine.

**Pull request merged**